zerver/lib/markdown/fenced_code.py

```diff
@@ -392,6 +395,24 @@ def format_code(self, lang: str, text: str) -> str:
         else:
             code = CODE_WRAP.format(langclass, self._escape(text))
 
+        # In order to display a "view-in-playground" option in the frontend,
+        # we need to know the language used in the codeblock. We tweak the HTML
+        # CodeHilite generates to add this language as a data-attribute.
+        if lang:
+            parsed_code = etree.HTML(code)
+            div_tag = parsed_code[0][0]
+            # We get the lexer subclass name instead of directly processing the lang, to avoid
+            # different tags being generated for each of the lang's alias. Eg: `js` and `javascript`
+            # would now be mapped to `JavaScript`. In case no lexer with that alias is found, we
+            # return back the text, wrapped in a data-codehilite tag.
+            try:
+                lexer_subclass_name = get_lexer_by_name(lang).name
+            except ClassNotFound:
+                lexer_subclass_name = lang
+            div_tag.attrib['data-codehilite-language'] = lexer_subclass_name
+            # Lxml implicitly converts tags like <span></span> into <span/>
+            # specifying method="c14n" when converting to string, prevents that.
+            code = etree.tostring(div_tag, method="c14n").decode()
```

Comment:
It would be kinda nice if the pygments offered a way to specify attributes (not just a list of CSS classes) to attach to the `pre` element being generated by `pygments`; this might require upstream contributions to both `pygments` and then `python-markdown`.  @andersk FYI in case you have ideas; I don't see a better way to do what we want with their current API.